engine/src/main/java/org/hibernate/validator/internal/engine/constraintvalidation/ConstraintTree.java

```diff
@@ -56,21 +62,27 @@
 
 	private final Type validatedValueType;
 
-	public ConstraintTree(ConstraintDescriptorImpl<A> descriptor, Type validatedValueType) {
-		this( descriptor, validatedValueType, null );
+	private volatile Optional<ConstraintValidator<A, ?>> constraintValidatorForDefaultConstraintValidatorFactory;
```

Comment:
What's the advantage of using `Optional`? It's not meant to be used as field really, using a nullable reference also saves the bytes of that wrapper.

Reply:
We need to make a difference between not yet defined and null. So `Optional` fits very well for this. And for the CHM, we have the same issue as we can't put null in the map.

Another solution is to use a dummy constraint validator, thus avoiding the wrapper but tbh I find it more readable with the `Optional`.

Reply:
Done.